src/cmd/winwatch.c

```diff
@@ -0,0 +1,543 @@
+                      MARGIN + (PAD + Dy(r)) * (i % rows))),
+                  screen->r.min);
+    draw(screen, insetrect(r, -1), lightblue, nil, ZP);
+}
+
+
+void drawwin(int i)
+{
+    draw(screen, win[i].r, lightblue, nil, ZP);
+    _string(screen, addpt(win[i].r.min, Pt(2, 0)), display->black, ZP,
+            font, win[i].label, nil, strlen(win[i].label),
+            win[i].r, nil, ZP, SoverD);
+    border(screen, win[i].r, 1, display->black, ZP);
+    win[i].dirty = 0;
+}
+
+
+int geometry(void)
+{
+    int i, ncols, z;
+    Rectangle r;
+
+    z = 0;
+    rows = (Dy(screen->r) - 2 * MARGIN + PAD) / (font->height + PAD);
+    if (rows * cols < nwin || rows * cols >= nwin * 2) {
+        ncols = nwin <= 0 ? 1 : (nwin + rows - 1) / rows;
+        if (ncols != cols) {
+            cols = ncols;
+            z = 1;
+        }
+    }
+
+    r = Rect(0, 0, (Dx(screen->r) - 2 * MARGIN + PAD) / cols - PAD,
+             font->height);
+    for (i = 0; i < nwin; i++)
+        win[i].r =
+            rectaddpt(rectaddpt
+                      (r,
+                       Pt(MARGIN + (PAD + Dx(r)) * (i / rows),
+                          MARGIN + (PAD + Dy(r)) * (i % rows))),
+                      screen->r.min);
+
+    return z;
+}
+
+
+void redraw(Image * screen, int all)
+{
+    int i;
+
+    all |= geometry();
+    if (all)
+        draw(screen, screen->r, lightblue, nil, ZP);
+    for (i = 0; i < nwin; i++)
+        if (all || win[i].dirty)
+            drawwin(i);
+    if (!all)
+        for (; i < onwin; i++)
+            drawnowin(i);
+
+    onwin = nwin;
+}
+
+
+void eresized(int new)
+{
+    if (new && getwindow(display, Refmesg) < 0)
+        fprint(2, "can't reattach to window");
+    geometry();
+    redraw(screen, 1);
+}
+
+
+void selectwin(XWindow win)
+{
+    XEvent ev;
+    long mask;
+
+    memset(&ev, 0, sizeof ev);
+    ev.xclient.type = ClientMessage;
+    ev.xclient.serial = 0;
+    ev.xclient.send_event = True;
+    ev.xclient.message_type = net_active_window;
+    ev.xclient.window = win;
+    ev.xclient.format = 32;
+    mask = SubstructureRedirectMask | SubstructureNotifyMask;
+
+    XSendEvent(dpy, root, False, mask, &ev);
+    XMapRaised(dpy, win);
+    XSync(dpy, False);
+}
+
+
+void click(Mouse m)
+{
+    int i, j;
+
+    if (m.buttons == 0 || (m.buttons & ~4))
+        return;
+
+    for (i = 0; i < nwin; i++)
+        if (ptinrect(m.xy, win[i].r))
+            break;
+    if (i == nwin)
+        return;
+
+    do
+        m = emouse();
+    while (m.buttons == 4);
+
+    if (m.buttons != 0) {
+        do
+            m = emouse();
+        while (m.buttons);
+        return;
+    }
+
+    for (j = 0; j < nwin; j++)
+        if (ptinrect(m.xy, win[j].r))
+            break;
+    if (j != i)
+        return;
+
+    selectwin(win[i].n);
+}
+
+
+void usage(void)
+{
+    fprint(2,
+           "usage: winwatch [-e exclude] [-W winsize] [-f font] [-n] [-s]\n");
+    exits("usage");
+}
+
+
+void main(int argc, char **argv)
```

Comment:
Use Plan 9 C style, where the function name always start on a new line. It makes it easier to find function definitions. Here, `g ^main` would tell me where all the main functions are in this directory.